src/signed-api-fetch/data-fetcher.ts

```diff
@@ -0,0 +1,304 @@
+import { clearInterval } from 'timers';
+import { go } from '@api3/promise-utils';
+import axios from 'axios';
+import { groupBy } from 'lodash';
+import { ethers } from 'ethers';
+import { Config } from '../config/schema';
+import { logger } from '../logger';
+import { DataStore, signedApiResponseSchema, SignedData } from '../types';
+import { localDataStore } from '../signed-data-store';
+
+// Express handler/endpoint path: https://github.com/api3dao/signed-api/blob/b6e0d0700dd9e7547b37eaa65e98b50120220105/packages/api/src/server.ts#L33
+// Actual handler fn: https://github.com/api3dao/signed-api/blob/b6e0d0700dd9e7547b37eaa65e98b50120220105/packages/api/src/handlers.ts#L81
+
+type PendingCall = {
+  fn: (deferredFunction: () => void) => Promise<void>;
+  nextRun: number;
+  lastRun: number;
+  running: boolean;
+  whoAmI: number;
+};
+
+type AirnodesAndSignedDataUrl = { urls: string[]; airnodeAddress: string; templateId: string };
+
+const HTTP_SIGNED_DATA_API_ATTEMPT_TIMEOUT = 10_000;
+const HTTP_SIGNED_DATA_API_HEADROOM = 1_000;
+
+let mainInterval: NodeJS.Timeout | undefined;
+let configRefresherInterval: NodeJS.Timeout | undefined;
+let dataFetcherInterval: NodeJS.Timeout | undefined;
+let config: Config | undefined;
+
+export const verifySignedData = (_signedData: SignedData) => {
+  // TODO https://github.com/api3dao/airseeker-v2/issues/23
+  // https://github.com/api3dao/airnode-protocol-v1/blob/5bf01edcd0fe76b94d3d6d6720b71ec658216436/contracts/api3-server-v1/BeaconUpdatesWithSignedData.sol#L26
+  return true;
+};
+
+// Useful for tests
+let axiosProd = axios;
+export const setAxios = (customAxios: any) => {
+  axiosProd = customAxios;
+};
+
+/**
+ * A function factory that outputs functions that call a remote signed data API.
+ *
+ * @param urls
+ * @param airnodeAddress
+ * @param whoAmI an identifier for this function so we can match up log entries
+ * @param dataStore the data store this function should call to store data
+ */
+const makeApiCallFn =
+  ({
+    urls,
+    airnodeAddress,
+    whoAmI,
+    dataStore,
+  }: Pick<AirnodesAndSignedDataUrl, 'urls' | 'airnodeAddress'> & {
+    whoAmI: number;
+    dataStore: DataStore;
+  }) =>
+  async (deferredFunction: () => void) => {
+    logger.debug(`[${whoAmI}] Started API call function`);
+
+    const url = urls[Math.ceil(Math.random() * urls.length) - 1];
+    logger.debug(`[${whoAmI}] Calling ${url}/${airnodeAddress}`);
+
+    const result = await go(
+      () =>
+        axiosProd({
+          method: 'get',
+          timeout: HTTP_SIGNED_DATA_API_ATTEMPT_TIMEOUT - HTTP_SIGNED_DATA_API_HEADROOM / 2,
+          //url: `${url}/${airnodeAddress}`,
+          url: `${url}/0xC04575A2773Da9Cd23853A69694e02111b2c4182`, // TODO override here so we can better simulate many apis
```

Comment:
I saw it while browsing the repositories the address here is not the address of the node. `0xc52EeA00154B4fF1EbbF8Ba39FDe37F1AC3B9Fd4` this is the one for Nodary Airnode address.

Reply:
Thanks @bdrhn9, this is just for testing, we ideally need a server that serves many airnodes with fresh data 😅

Do you have other airnode addresses on this server?

Reply:
> Do you have other airnode addresses on this server? Also it looks like the timestamps don't change.
Yeah there is one belongs to Nodary `0xc52EeA00154B4fF1EbbF8Ba39FDe37F1AC3B9Fd4` and it is being updated freshly.